cmake/nlohmann_json/Findnlohmann_json.cmake

```diff
@@ -0,0 +1,15 @@
+include(FindPackageHandleStandardArgs)
+set(${CMAKE_FIND_PACKAGE_NAME}_CONFIG ${CMAKE_CURRENT_LIST_FILE})
+find_package_handle_standard_args(nlohmann_json CONFIG_MODE)
```

Comment:
`nlohmann_json`?! It seems it's coming from cmake output directory. My suggestion is to change it.

Reply:
What are the reasons to change it? What alternative do you suggest?

Reply:
@acloudiator Can you please be more specific? It would be nice to merge this big commit in order to move on. Thanks!

Reply:
@frisso Following my conversation with @mauriciovasquezbernal this makes more sense and we've it under src/libs. I'm marking this conversation as resolved but still wait for @goldenrye feedback as this is a huge PR.